src/instructlab/configuration.py

```diff
@@ -312,47 +333,84 @@ def fill_defaults(
 class _serve_vllm(BaseModel):
     """Class describing configuration of vllm serving backend."""
 
-    llm_family: str = ""
-    max_startup_attempts: int | None = None
-    gpus: Optional[int] = None
+    llm_family: str = Field(
+        default="",  # TODO: convert to None and use a pattern to validate
+        description="Large Language Model Family",
+        examples=["merline", "granite"],
+    )
+    max_startup_attempts: int | None = Field(
+        default=None,
+        description="Maximum number of attempts to start the vLLM server.",
+    )
+    gpus: Optional[int] = Field(default=None, description="Number of GPUs to use.")
     # arguments to pass into vllm process
-    vllm_args: list[str] | None = Field(default_factory=lambda: [])
+    vllm_args: list[str] | None = Field(
+        default_factory=list,
+        description="vLLM specific arguments. All settings can be passed as a list of strings, see: https://docs.vllm.ai/en/v0.5.2/serving/openai_compatible_server.html#command-line-arguments-for-the-server.",
+        examples=[
+            ["--dtype", "auto"],
+            ["--lora-alpha", "32"],
+        ],
+    )
 
 
 class _serve_llama_cpp(BaseModel):
     """Class describing configuration of llama-cpp serving backend."""
 
-    gpu_layers: int = -1
-    max_ctx_size: PositiveInt = 4096
-    llm_family: str = ""
+    gpu_layers: int = Field(
+        default=-1,
+        description="Number of model layers to offload to GPU. -1 means all layers.",
+    )
+    max_ctx_size: PositiveInt = Field(
+        default=4096,
+        description="Maximum number of tokens that can be processed by the model.",
+    )
+    llm_family: str = Field(
+        default="",  # TODO: convert to None and use a pattern to validateè !q
+        description="Large Language Model Family",
+        examples=["merline", "granite"],
+    )
 
 
 class _serve(BaseModel):
     """Class describing configuration of the serve sub-command."""
 
     # model configuration
     model_config = ConfigDict(extra="ignore", protected_namespaces=())
-
     # vllm configuration
-    vllm: _serve_vllm = _serve_vllm(
-        llm_family="",
-        max_startup_attempts=300,
+    vllm: _serve_vllm = Field(
+        default_factory=lambda: _serve_vllm,
+        description="vLLM serving settings.",
     )
-
     # llama-cpp configuration
-    llama_cpp: _serve_llama_cpp = _serve_llama_cpp(
-        gpu_layers=-1,
-        max_ctx_size=4096,
-        llm_family="",
+    llama_cpp: _serve_llama_cpp = Field(
+        default_factory=lambda: _serve_llama_cpp,
+        description="llama-cpp serving settings.",
     )
-
     # required fields
```

Comment:
These lines about required fields should now be removed - there are no longer any "required" fields, all have defaults.